go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -413,11 +413,23 @@ func (c *nadController) GetActiveNetworkForNamespace(namespace string) (util.Net
 	if !util.IsNetworkSegmentationSupportEnabled() {
 		return &util.DefaultNetInfo{}, nil
 	}
+
+	// check if required UDN label is on namespace
+	ns, err := c.namespaceLister.Get(namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get namespace %q: %w", namespace, err)
+	}
+	if _, exists := ns.Labels[types.RequiredUDNNamespaceLabel]; !exists {
+		// UDN required label not set on namespace, assume default network
+		return &util.DefaultNetInfo{}, nil
```

Comment:
is it always correct to assume not having the label means default net? because semantically speaking if label is missing and CRs are present there are cases where we block pod creation and don't make the pod part of default network right? do we need an unknown state or at least in how we use GetActiveNamespace we are OK ?
I guess it could be that I am confusing what it means for "a pod to be part of UDN namespace" v/s "when a namespace is considered for udn or not"

Reply:
what I mean is not having a NAD + not having a UDN + not having CUDN + not having an annotation is what should be considered as "defaultnet" right?

Reply:
"is it always correct to assume not having the label means default net?"

that's what we decided to go with. It means default net for the pod, but the UDN will report error.

Reply:
hmm I see will ping you a bit on Monday to talk about this